src/util/TokenService.js

```diff
@@ -105,25 +102,56 @@ class TokenService {
                 typeof tokenOrTokenFunction !== 'function') ||
             !ids.every((itemId) => itemId.startsWith(TYPED_ID_FOLDER_PREFIX) || itemId.startsWith(TYPED_ID_FILE_PREFIX))
         ) {
-            return Promise.reject(error);
+            throw error;
         }
 
         // Token is a simple string or null or undefined
         if (!tokenOrTokenFunction || typeof tokenOrTokenFunction === 'string') {
-            return Promise.resolve(createIdTokenMap(ids, tokenOrTokenFunction));
+            return createIdTokenMap(ids, tokenOrTokenFunction);
         }
 
         // Token is a function which returns a promise
         // that on resolution returns an id to token map.
-        return new Promise(async (resolve: Function, reject: Function) => {
-            // $FlowFixMe Already checked null above
-            const token = await tokenOrTokenFunction(ids);
-            if (!token || typeof token === 'object' || typeof token === 'string') {
-                resolve(createIdTokenMap(ids, token));
-            } else {
-                reject(error);
-            }
-        });
+        const token = await tokenOrTokenFunction(ids);
+        if (!token || typeof token === 'object' || typeof token === 'string') {
```

Comment:
Why do we have to support a single string token coming back here? Shouldn't `getToken` be called in that case?

Reply:
I am gonna nuke bunch of the code here, not really needed